modules/replace-the-certificate-authority-clientca.adoc

```diff
@@ -0,0 +1,77 @@
+// Module included in the following assemblies:
+//
+// * security/certificates/api-server.adoc
+
+:_content-type: PROCEDURE
+[id="replace-the-certificate-authority_{context}"]
+
+= Replace the installer-generated kubeconfig before replacing it with a newly generated CA certificate
+
+The installer-generated kubeconfig cannot be removed, but it can be invalidated and replaced with a newly generated CA certificate.
+
+You may choose to invalidate the installer-generated kubeconfig, stored in the `admin-kubeconfig-client-ca` configmap under the `openshift-config` namespace, when:
+
+* You don't trust who installed the cluster
+* The kubeconfig is leaked
+* Other security-related needs exist, such as periodic rotation of the kubeconfig
+
+[IMPORTANT]
+====
+To avoid being locked out of the cluster, have an alternative way to login, such as, using an OAuth-authenticated administrator user or using a client certificate signed by an additional client CA.
+====
+
+.Procedure
+
+. Optional: Generate a new self-signed CA, unless an existing corporate or other CA is to be used.
++
+[source,terminal]
+----
+$ export NAME="custom"
+$ export CA_SUBJ="/OU=openshift/CN=admin-kubeconfig-signer-custom"
+
+# set the CA validity to 10 years
+$ export VALIDITY=3650
+
+# generate the CA private key
+$ openssl genrsa -out ${NAME}-ca.key 4096
+
+# Create the CA certificate
+$ openssl req -x509 -new -nodes -key ${NAME}-ca.key -sha256 -days $VALIDITY -out ${NAME}-ca.crt -subj "${CA_SUBJ}"
+----
++
+. Generate a new `system:admin` certificate. The client certificate must have the user into the x.509 subject CN field and the group into the O field.
```

Comment:
| The client certificate must have the user into the x.509 subject CN field and the group into the O field.

Should these be call-outs, pointing out where the values go?

Reply:
I'd say no, as then we'd be documenting the certificate which would be a third party item.

On the other hand this sentence doesn't make sense to me, so I'll try to find out how it can be addressed.